conceptual/Aspects/Diagnostics.md

```diff
@@ -55,6 +57,11 @@ To suppress a diagnostic:
 
 The following logging aspect requires a `_logger` field to exist, but it is likely that this field will never be used in user code but only in generated code. Because the IDE does not see the generated code, it will report the `CS0169` warning, which is misleading and annoying to the user. The aspect suppresses this warning.
 
+<!--- and then to add insult to injury you go on to show me a way to surpress a warning I likely wouldn't have created given that a fabric is the obvious way to do logging.
+
+NOW OF MUCH MORE USE  consider the following.  Most generic logging is going to be tell me I've entered a method with x params, I'm leaving a method with x result and tell me if we hit an exception.
+what you can't really predict in advance is whether for example there will be something like a for loop or a dictionary in a method.  Now lets say that a team's rule is that all for loops in financial 
+methods are logged, now thats where an example of this sort would really come into it's own becuase you'd be demonstrating something that might well have to be done.  -->
```

Comment:
```suggestion
```

This use case cannot be easily implemented because the public API does not give you access to the method body.